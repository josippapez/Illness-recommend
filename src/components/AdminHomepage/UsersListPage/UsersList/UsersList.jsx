import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Container, Table } from 'reactstrap';
import { useDispatch, useSelector } from 'react-redux';
import ListHeader from '../../../SharedComponents/ListHeader/ListHeader';
import '../../../../styles/ListTable.scss';
import { userInfoFetched } from '../../../../store/actions/usersActions';

const UsersList = props => {
  const currentUser = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [headers, setHeaders] = useState([
    {
      header: 'Ime i prezime',
      headerKey: 'name',
    },
    {
      header: 'Email',
      headerKey: 'email',
    },
    {
      header: 'Rola',
      headerKey: 'role',
    },
    {
      header: 'Akcije',
    },
  ]);
  return (
    <table style={{ width: '100%' }} className="list-table">
      <ListHeader setHeaders={setHeaders} headers={headers} />
      <tbody className="list-table__item-row">
        {props.usersList.map(user => {
          return (
            user.id !== currentUser.id && (
              <tr className="spacer  item-row" key={user.id}>
                <td>{user.name}</td>
                <td>{user.email}</td>
                <td>{user.role}</td>
                <td>
                  <button
                    id="link-to-student-page"
                    onClick={() => {
                      dispatch(userInfoFetched({ data: null }));
                      props.setUserId(user.id);
                      props.setShowUserInfoModal(true);
                    }}
                  >
                    Odaberi
                  </button>
                </td>
              </tr>
            )
          );
        })}
      </tbody>
    </table>
  );
};

UsersList.propTypes = {
  usersList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  setUserId: PropTypes.func,
  setShowUserInfoModal: PropTypes.func,
};

export default UsersList;
