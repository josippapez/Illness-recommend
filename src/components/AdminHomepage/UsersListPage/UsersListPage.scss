.navigation-setup:is(.darkTheme) {
  .users-list-page {
    &__list {
      &__display-list {
        &__not-found {
          color: var(--white);
        }
      }
    }
  }
}

.users-list-page {
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  &__list {
    &__display-list {
      &__not-found {
        font-family: "Lato, Bold";
        font-size: 18px;
        width: fit-content;
        margin: auto;
      }
    }
  }
}
