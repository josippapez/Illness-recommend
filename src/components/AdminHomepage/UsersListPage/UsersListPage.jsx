import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './UsersListPage.scss';
import { useDispatch, useSelector } from 'react-redux';
import { getAllUsers, searchUsersByText } from '../../../store/actions';
import DataDisplay from '../../SharedComponents/DataDisplay/DataDisplay';
import UsersList from './UsersList/UsersList';
import UserInfoModal from './UserInfoModal/UserInfoModal';
import Search from '../../SharedComponents/Search/Search';

const UsersListPage = props => {
  const dispatch = useDispatch();
  const usersList = useSelector(state => state.usersList);

  const [showUserInfoModal, setShowUserInfoModal] = useState(false);
  const [userId, setUserId] = useState(null);
  const [searchQuery, setSearchQuery] = useState(null);
  useEffect(() => {
    dispatch(getAllUsers());
  }, []);

  return (
    <div className="users-list-page">
      <DataDisplay
        dataHeader="Popis korisnika"
        headerBolded
        headerFontSize={23}
        headerTextColor={props.theme.darkTheme ? '#fff' : '#005BA7'}
        dataFullWidth
        TopSpacing={30}
        floatDataRight
        data={
          <Search
            searchingByInfo="*Pretraživanje po Imenu i Email-u"
            fetchData={() => {
              return getAllUsers();
            }}
            fetchDataByName={text => {
              return searchUsersByText(text);
            }}
            setSearchQuery={setSearchQuery}
          />
        }
      />
      <DataDisplay
        TopSpacing={40}
        dataFullWidth
        data={
          <div className="users-list-page__list__display-list">
            {usersList && usersList.users && usersList.users.length > 0 ? (
              <UsersList
                usersList={usersList.users}
                setUserId={setUserId}
                setShowUserInfoModal={setShowUserInfoModal}
              />
            ) : (
              <div className="users-list-page__list__display-list__not-found">
                {searchQuery
                  ? 'Nema pronđenih korisnika'
                  : 'Nema unesenih korisnika'}
              </div>
            )}
          </div>
        }
      />
      {showUserInfoModal && (
        <UserInfoModal
          userId={userId}
          setUserId={setUserId}
          setShowUserInfoModal={setShowUserInfoModal}
        />
      )}
    </div>
  );
};

UsersListPage.propTypes = {
  theme: PropTypes.object,
};

export default UsersListPage;
