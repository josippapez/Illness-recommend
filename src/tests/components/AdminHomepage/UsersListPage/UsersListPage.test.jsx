/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import UsersList from '../../../../components/AdminHomepage/UsersListPage/UsersList/UsersList';
import {
  fetchUserInfoById,
  getAllAlergies,
  getAllUsers,
  searchUsersByText,
  userInfoFetched,
} from '../../../../store/actions';
import UsersListPage from '../../../../components/AdminHomepage/UsersListPage/UsersListPage';

jest.mock('../../../../store/actions');

const mockStore = configureStore();
const store = mockStore({
  user: {
    id: 4,
    name: 'Josip PapeÅ¾',
    email: 'tt@tt.com',
    role: 'admin',
    isAuthenticated: true,
    error: null,
    status: null,
    message: null,
    userInfo: {
      data: {
        id: null,
        name: null,
        email: null,
        role: null,
      },
      error: null,
      status: null,
    },
  },
  alergies: {
    alergies: [
      {
        id: 1,
        name: 'Amikacin',
      },
      {
        id: 2,
        name: 'Azitromicin',
      },
    ],
    error: null,
    status: null,
  },
  usersList: {
    users: [
      {
        id: 1,
        email: 'test@tt.com',
        name: 'Anno Anno',
        role: 'user',
      },
      {
        id: 5,
        email: 'a@tt.com',
        name: 'Crona-Mante',
        role: 'user',
      },
    ],
    error: null,
    status: null,
  },
});

const props = {
  theme: {
    darkTheme: false,
  },
};

describe('Users list page', () => {
  let component;

  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'root');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);

  beforeEach(() => {
    getAllUsers.mockImplementation(() => ({ type: 'test' }));
    searchUsersByText.mockImplementation(() => ({ type: 'test' }));
    userInfoFetched.mockImplementation(() => ({ type: 'test' }));
    getAllAlergies.mockImplementation(() => ({ type: 'test' }));
    fetchUserInfoById.mockImplementation(() => ({ type: 'test' }));
    component = mount(
      <Provider store={store}>
        <UsersListPage {...props} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('modal opens on user select button', () => {
    component.find('#link-to-medication-page').first().simulate('click');
    expect(userInfoFetched).toHaveBeenCalled();
  });
});
