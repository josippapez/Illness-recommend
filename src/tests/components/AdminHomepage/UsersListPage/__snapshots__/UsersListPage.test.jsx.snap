// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users list page renders correctly 1`] = `"<div class=\\"users-list-page\\"><div style=\\"margin-top: 30px; margin-bottom: 0px;\\"><div class=\\"row-display\\"><div class=\\"dataDisplayHeader col label bolded\\" style=\\"font-weight: bold; color: rgb(0, 91, 167); transition: color 0.2s ease-in-out; width: 150px; min-width: 25%; margin-right: 16px; font-size: 23px; text-align: left;\\">Popis korisnika</div><div class=\\"col\\" style=\\"width: 100%; display: flex; justify-content: flex-end;\\"><div class=\\"search\\"><form class=\\"search__form\\"><input type=\\"text\\" placeholder=\\"Pretraživanje\\" class=\\"search-input\\" value=\\"\\"></form><small class=\\"search-by-text\\">*Pretraživanje po Imenu i Email-u</small></div></div></div></div><div style=\\"margin-top: 40px; margin-bottom: 0px;\\"><div class=\\"row-display\\"><div class=\\"dataDisplayHeader col label\\" hidden=\\"\\" style=\\"color: black; transition: color 0.2s ease-in-out; width: 150px; min-width: 25%; margin-right: 16px; font-size: 16px; text-align: left;\\"></div><div class=\\"col\\" style=\\"width: 100%;\\"><div class=\\"users-list-page__list__display-list\\"><table style=\\"width: 100%;\\" class=\\"list-table\\"><thead class=\\"list-table__table-header\\"><tr class=\\"list-table__table-header__row\\"><th class=\\"list-table__table-header__row__header\\" id=\\"name\\">Ime i prezime<span></span></th><th class=\\"list-table__table-header__row__header\\" id=\\"email\\">Email<span></span></th><th class=\\"list-table__table-header__row__header\\" id=\\"role\\">Rola<span></span></th><th class=\\"list-table__table-header__row__header\\">Akcije<span></span></th></tr></thead><tbody class=\\"list-table__item-row\\"><tr class=\\"spacer  item-row\\"><td>Anno Anno</td><td>test@tt.com</td><td>user</td><td style=\\"width: 200px; min-width: 50px;\\"><button id=\\"link-to-medication-page\\">Odaberi</button></td></tr><tr class=\\"spacer  item-row\\"><td>Crona-Mante</td><td>a@tt.com</td><td>user</td><td style=\\"width: 200px; min-width: 50px;\\"><button id=\\"link-to-medication-page\\">Odaberi</button></td></tr></tbody></table></div></div></div></div></div>"`;
